pyvista/core/utilities/misc.py

```diff
@@ -92,13 +92,12 @@ def abstract_class(cls_):  # numpydoc ignore=RT01
         The class to be decorated as abstract.
 
     """
+    # def __new__(cls, *args, **kwargs):
+    #     if cls is cls_:
+    #         raise TypeError(f'{cls.__name__} is an abstract class and may not be instantiated.')
+    #     return object.__new__(cls)
 
-    def __new__(cls, *args, **kwargs):
-        if cls is cls_:
-            raise TypeError(f'{cls.__name__} is an abstract class and may not be instantiated.')
-        return object.__new__(cls)
-
-    cls_.__new__ = __new__
+    # cls_.__new__ = __new__
     return cls_
```

Comment:
>What was the use case for this?

I think the goal here is simply to prevent end-users from trying to instantiate these classes.

> We could define an abstract __init__(), the only restriction is that concrete child classes would have to do at least something like

This might work... I tried a few similar approaches but I think what you propose would work. I'm going to ask that we delay figuring this out in its own PR so that we can merge and release these changes to prevent the complete incompatibility with VTK

Reply:
I've looked a bit into this, the decorator was added in https://github.com/pyvista/pyvista/pull/752 but its predecessor `__new__` logic was already added in https://github.com/pyvista/pyvista/pull/274 (which might not be the first instance of the pattern either). From the looks of it this was just a short-hand for having to actually bother with `@abstractmethods`. So coming up with a replacement mechanism that uses `@abstractmethod` might be completely pointless.

We have plenty of classes decorated with this, but e.g. `BaseReader` is an example where many/most subclasses only override class attributes, but don't define any methods. In this case it would be a pain in the ass to define a no-op `__init__()` in each of them (which is not to say that it might not be the cleanest course of action, just that I don't see an easy way out if we have to leave `__new__` alone).

Reply:
OK, I think I've fixed it in https://github.com/pyvista/pyvista/pull/5650/commits/efeb4937a63652ccd99915d8fdc7342be840e6a4. The issue was that we were jumping too much ahead of the MRO.

Let's see if CI is happy...

Reply:
CI seems happy here, thanks for figuring this out!